chatrooms.lua

```diff
@@ -94,13 +145,39 @@ function ToArrival( tUser, sMessage )
 		return true
 	else
 		local sChat = sMessage:match "%b$$(.*)|"
-		SendToRoom( tUser, sTo, sChat )
+		SaveToFile( sChat, sTo )
+		local sCmd, sData = sMessage:match "%b$$%b<>%s+[-+*/?!#](%w+)%s?(.*)|"
+		if not sCmd then
+			SendToRoom( tUser, sTo, sChat )
+		elseif sCmd:lower() == "history" then
```

Comment:
Instead of computing `sCmd:lower()` everytime, break the if-else block:

``` Lua
if not sCmd then ... end
sCmd = sCmd:lower()
-- further processing
```